helix-core/src/main/java/org/apache/helix/HelixCloudProperties.java

```diff
@@ -0,0 +1,108 @@
+package org.apache.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.InputStream;
+import java.util.Properties;
+
+import org.apache.helix.model.CloudConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Hold helix cloud properties coming from different sources.
+ */
+public class HelixCloudProperties {
+  private static final Logger LOG = LoggerFactory.getLogger(HelixCloudProperties.class.getName());
+
+  private final Properties _properties = new Properties();
+
+  /**
+   * Initialize helix cloud properties
+   * To combine cloud properties from different sources, firstly, the cloud config will be examined, if the provider has its
+   * properties defined in Helix, then the corresponding file under helix-core/src/main/resources/ will be read. Finally,
+   * user input properties will be merged. If there is any duplicate key existing, the user defined property will have
+   * higher priority and override other properties
+   * @param userProperties
+   * @param cloudConfig
+   */
+  public HelixCloudProperties(Properties userProperties, CloudConfig cloudConfig) {
+    switch (cloudConfig.getCloudProvider()) {
+      case "AZURE":
```

Comment:
Question: Has this been formatted properly? I believe there shouldn't be an indentation between switch and case?

Reply:
This file was removed.